lamindb/models/artifact.py

```diff
@@ -2142,23 +2148,30 @@ def replace(
         self._old_suffix = self.suffix
 
     def open(
-        self, mode: str = "r", is_run_input: bool | None = None, **kwargs
-    ) -> Union[
-        AnnDataAccessor,
-        BackedAccessor,
-        SOMACollection,
-        SOMAExperiment,
-        SOMAMeasurement,
-        PyArrowDataset,
-    ]:
+        self,
+        mode: str = "r",
+        df_engine: Literal["pyarrow", "polars"] = "pyarrow",
```

Comment:
Can we call this `engine` or `backend` or another simpler word?

Even if we have "non-df engines" in the future (whatever they will be), I don't think we want to confront the user with a list of prefixed `_engine` arguments.

Collections will in all cases in which `open()` makes sense be homogeneous and we'll never have a need to mix `df_engine` with other parameters.

Reply:
We might be introducing a `DataFrameCollection` and a `ZarrCollection` or things like this in the future.

And then the engine for dataframes could be pyarrow or polars and for zarr it could be kerchunk.

Of course I can't predict specifics etc. but `df_engine` would to me appear like a "patch" where "we get something to work for dataframes" rather than sticking with the original vision to create a cohesive abstraction for a bunch of formats.

Reply:
And we might eliminate the `df` abbreviation from the API at some point in favor of `to_pandas`, `pandas`, `polars`, `parquet`, `pyarrow`, etc. -- sometimes one might call this a "table" (e.g. in pyarrow there are no "dataframes" I think).